package/src/utils/DBSyncManager.ts

```diff
@@ -0,0 +1,119 @@
+import type { StreamChat } from 'stream-chat';
+
+import { executePendingTasks } from './pendingTaskUtils';
+
+import { handleEventToSyncDB } from '../components/Chat/hooks/handleEventToSyncDB';
+import { getAllChannelIds, getLastSyncedAt, upsertLastSyncedAt } from '../store/apis';
+import { QuickSqliteClient } from '../store/QuickSqliteClient';
+
+import type { PreparedQueries } from '../store/types';
+
+/**
+ * DBSyncManager has the responsibility to sync the channel states
+ * within local database whenever possible.
+ *
+ * Components can get the current sync status using DBSyncManager.getCurrentStatus().
+ * Or components can attach a listener for status change as following:
+ *
+ * ```tsx
+ * useEffect(() => {
+ *  const unsubscribe = DBSyncManager.onSyncStatusChange((syncStatus) => {
+ *    if (syncStatus) {
+ *      doSomething();
+ *    }
+ *  })
+ *
+ *  return () => unsubscribe();
+ * })
+ * ```
+ */
+export class DBSyncManager {
+  static syncStatus = false;
```

Comment:
I didn't expect that `syncStatus` will be `boolean`, because of it's name I expected an enum str or such.
when true/false what does it say about the syncing? has it occurred, started, happening atm etc?